### PR TITLE
PR: Enhance "colour.models.rgb.transfer_functions.log" module. 

### PR DESCRIPTION
**Pull request by njwardhan**
This pull request enhances the `log.py` module by adding the following **log** node functions from the CLF 3.0 specifications:

* `logarithm_basic`
* `logarithm_lin_to_log`
* `logarithm_log_to_lin`
* `logarithm_camera_lin_to_log`
* `logarithm_camera_log_to_lin`

### CODE REVIEW

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  4
         

Clones added
============
- colour/models/rgb/transfer_functions/log.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5799134&bid=18981785)

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  4
         

Clones added
============
- colour/models/rgb/transfer_functions/log.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5799134&bid=18981785)

**Comment by njwardhan** [comment]
@KelSolaar, would request you to review this PR and let me know of the required changes. Thankyou!

**Comment by KelSolaar** [comment]
LGTM at first glance, there are only a few minor things I noted, the main thing I would like to discuss is the naming though.

I'm not a huge fan for the current names:

- `logarithm_basic`
- `logarithm_lin_to_log`
- `logarithm_log_to_lin`
- `logarithm_camera_lin_to_log`
- `logarithm_camera_log_to_lin`

For consistency with the `exponent_function_basic` and `exponent_function_monitor_curve`, we would probably need to:

- Change the `logarithm` to `logarithmic_function`
- Replace the `logarithm_lin_to_log`, `logarithm_log_to_lin`, `logarithm_camera_lin_to_log` and `logarithm_camera_log_to_lin` definitions with a single `logarithmic_function_camera` akin to `exponent_function_monitor_curve`.

What do you think @nick-shaw ?

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5799134&bid=18981785)

**Comment by njwardhan** [comment]
@KelSolaar, @nick-shaw, the latest commit improves the nomenclature used in the PR and brings it down to two major functions:

- `logarithm_function_basic`
- `logarithm_function_camera`

Would love to know your thoughts and suggestions regarding the same. Thankyou!

File: colour/models/rgb/transfer_functions/log.py [hunk 1]
```diff
@@ -40,7 +50,293 @@
...
+        It is the slope (or gain) applied to the log side
+        of the logarithmic segment. Its default value is 1.
+    lin_side_slope : numeric, optional
+        It is the slope of the linear side
+        of the logarithmic segment. Its default value is 1.
+    log_side_offset : numeric, optional
+        It is the offset applied to the log side
+        of the logarithmic segment. Its default value is 0.
+    lin_side_offset : numeric, optional
+        It is the offset applied to the linaer side
```
Comment (nick-shaw) [inline colour/models/rgb/transfer_functions/log.py]: Typo in spelling of "linear" here.

**Comment by nick-shaw** [comment]
It appears that `logarithm_function_basic` includes a `base` parameter which is unused, since there is no option to use an arbitrary base. The available styles only include bases 10 and 2. At least that's how it seems to me.

File: colour/models/rgb/transfer_functions/log.py (hunk 1 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py]: Will rectify it. Thankyou!

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5799134&bid=18981785)

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5799134&bid=18981785)

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5799134&bid=18981785)

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5799134&bid=18981785)

**Comment by KelSolaar** [comment]
Thanks @njwardhan, merging!

**Pull request merged**